mixins/localize-mixin.md

```diff
@@ -0,0 +1,294 @@
+# LocalizeMixin
+
+The `LocalizeMixin` adds language resolution, timezone and locale overrides support.
+
+## Automatic Language, Timezone and Override Support
+
+The user's language, timezone and any D2L locale overrides are automatically fetched by `LocalizeMixin` from the `<html>` element's `lang`, `data-timezone` and `data-intl-overrides` attributes respectively.
+
+## Usage
+
+To use `LocalizeMixin`, apply the mixin then implement the following functions (examples in "Language Resources"):
+* `getLanguage(langs)`: returns the page's language based upon `langs`, an array of possible languages determined by `LocalizeMixin` (e.g., `['ar-dz', 'ar', 'en']`)
+* `async getLangResources(lang)`: retrieves the available language resources for `lang`
+
+Consume your web component in a page which has the lang attribute set on the <html> element:
+```html
+<html lang="ar">
+	<head>
+		<script type="module">
+			import '../my-element.js';
+		</script>
+	</head>
+	<body>
+		<my-elem></my-elem>
+	</body>
+</html>
+```
+
+If the language of the page changes (via an update to the `lang` attribute on `<html>`), the mixin will automatically detect the change and re-render the web component. It will fire the event `d2l-localize-behavior-language-changed` when this occurs.
+
+## Language Resources
+
+* Always provide entries for base languages (e.g. "en", "fr", "pt") so that if the user is using a regional language (e.g. "en-gb", "fr-ca", "pt-br") which is missing, it can fall back to the base language.
+* If there's no entry for a particular language, and no base language, the value of `data-lang-default` on the `<html>` element will be used.
+* If no `data-lang-default` is specified, "en" will be used as a last resort.
+
+### Resources Property Example
+
+In this example the language resources for `ar` and `en` are stored in the `__langResources` object.
+
+```javascript
+import { LocalizeMixin } from '@brightspace-ui/core/mixins/localize-mixin.js';
+class MyComponent extends LocalizeMixin(LitElement) {
+	static get properties() {
+		return {
+			__langResources: { type: Object }
+		};
+	}
+
+	constructor() {
+		super();
+		this.__langResources = {
+			'ar': {
+				more: 'المزيد',
+				less: 'أقل'
+			},
+			'en': {
+				more: 'more',
+				less: 'less'
+			}
+		};
```

Comment:
In this example, technically I don't think the language resources need to be in a private property right? Like they could have been defined in a constant somewhere in the file? Just thinking of a case where everyone copies this example and then with class inheritance you could have a base class and an inherited class both doing this and clobbering each other's language resources.

Reply:
Could we still use a constant defined outside the class (in the same file) instead of a local private property? That way I think we avoid redefining it for every instance of the class that gets created (less memory).